lib/view_component/base.rb

```diff
@@ -271,56 +288,58 @@ def content_evaluated?
+    mattr_accessor :view_component_path, instance_writer: false, default: "app/components"
 
-    # Always generate translations file alongside the component:
+    # Parent class for generated components
     #
-    #     config.view_component.generate_locale = true
+    #     config.view_component.component_parent_class = "MyBaseComponent"
     #
-    # Defaults to `false`.
+    # Defaults to nil. If this is falsy, generators will use
+    # "ApplicationComponent" if defined, "ViewComponent::Base" otherwise.
     #
-    mattr_accessor :generate_locale, instance_writer: false, default: false
+    mattr_accessor :component_parent_class, instance_writer: false
 
-    # Always generate as many translations files as available locales:
+    # Configuration for generators.
+    #
+    # All options under this namespace default to `false` unless otherwise
+    # stated.
     #
-    #     config.view_component.generate_distinct_locale_files = true
+    # #### #sidecar
     #
-    # Defaults to `false`.
+    # Always generate a component with a sidecar directory:
     #
-    # One file will be generated for each configured `I18n.available_locales`.
-    # Fallback on `[:en]` when no available_locales is defined.
+    #     config.view_component.generate.sidecar = true
     #
-    mattr_accessor :generate_distinct_locale_files, instance_writer: false, default: false
-
-    # Path for component files
+    # #### #stimulus_controller
     #
-    #     config.view_component.view_component_path = "app/my_components"
+    # Always generate a Stimulus controller alongside the component:
     #
-    # Defaults to `app/components`.
+    #     config.view_component.generate.stimulus_controller = true
     #
-    mattr_accessor :view_component_path, instance_writer: false, default: "app/components"
-
-    # Parent class for generated components
+    # #### #locale
     #
-    #     config.view_component.component_parent_class = "MyBaseComponent"
+    # Always generate translations file alongside the component:
     #
-    # Defaults to "ApplicationComponent" if defined, "ViewComponent::Base" otherwise.
+    #     config.view_component.generate.locale = true
     #
-    mattr_accessor :component_parent_class, instance_writer: false
-
-    # Always generate a component with a sidecar directory:
+    # #### #distinct_locale_files
+    #
+    # Always generate as many translations files as available locales:
     #
-    #     config.view_component.generate_sidecar = true
+    #     config.view_component.generate.distinct_locale_files = true
     #
-    # Defaults to `false`.
+    # One file will be generated for each configured `I18n.available_locales`,
+    # falling back to `[:en]` when no available_locales is defined.
     #
-    mattr_accessor :generate_sidecar, instance_writer: false, default: false
+    mattr_accessor :generate, instance_writer: false, default: (ActiveSupport::OrderedOptions.new(false).tap do |c|
+      c.component_path = "app/components"
```

Comment:
Shouldn't this reuse `config.view_component.view_component_path`?

Reply:
I don't think `generate` needs the `component_path` option at all - I'd have thought it got removed when I dropped ceed17032b57346ce8eff5bf302c834b403ee1f1.